modules/Bio/EnsEMBL/Compara/RunnableDB/PairAligner/PairAlignerStats.pm

```diff
@@ -61,24 +96,16 @@ use warnings;
 use Bio::EnsEMBL::Hive::DBSQL::DBConnection;
 use Bio::EnsEMBL::Hive::Utils 'stringify';  # import 'stringify()'
 use Bio::EnsEMBL::Utils::URI;
+
 use Bio::EnsEMBL::Compara::Utils::CoreDBAdaptor;
 
 use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
 
-=head2 fetch_input
-
-  Implementation of the Bio::EnsEMBL::Hive::Process interface
-
-=cut
 
 sub fetch_input {
   my ($self) = @_;
 
   return if ($self->param('skip'));
-  #Default directory containing bed files.
-  if (!defined $self->param('bed_dir')) {
-      die ("Must define a location to dump the bed files using the parameter 'bed_dir'");
-  }
```

Comment:
So you prefer to have it here than just `$self->param_required()` at line 221 (when the code has already run partially). Sounds reasonable.

Reply:
Yes. Ideally, eHive would have a mechanism to declare the mandatory parameters and they could be checked before starting the job (even before creating the job ?) but it's not the case. It's just to avoid wasting some time